File: 012-IntegerToRoman.java
class Solution {
  public String intToRoman(int num) {
    StringBuilder out = new StringBuilder(30);

    int quotient = num / 1000;
    for (int i = 0; i < quotient; i++) {
      out.append('M');
      num -= 1000;
    }

    quotient = num / 900;
    for (int i = 0; i < quotient; i++) {
      out.append('C');
      out.append('M');
      num -= 900;
    }

    quotient = num / 500;
    for (int i = 0; i < quotient; i++) {
      out.append('D');
      num -= 500;
    }

    quotient = num / 400;
    for (int i = 0; i < quotient; i++) {
      out.append('C');
      out.append('D');
      num -= 400;
    }

    quotient = num / 100;
    for (int i = 0; i < quotient; i++) {
      out.append('C');
      num -= 100;
    }

    quotient = num / 90;
    for (int i = 0; i < quotient; i++) {
      out.append('X');
      out.append('C');
      num -= 90;
    }

    quotient = num / 50;
    for (int i = 0; i < quotient; i++) {
      out.append('L');
      num -= 50;
    }

    quotient = num / 40;
    for (int i = 0; i < quotient; i++) {
      out.append('X');
      out.append('L');
      num -= 40;
    }

    quotient = num / 10;
    for (int i = 0; i < quotient; i++) {
      out.append('X');
      num -= 10;
    }

    quotient = num / 9;
    for (int i = 0; i < quotient; i++) {
      out.append('I');
      out.append('X');
      num -= 9;
    }

    quotient = num / 5;
    for (int i = 0; i < quotient; i++) {
      out.append('V');
      num -= 5;
    }

    quotient = num / 4;
    for (int i = 0; i < quotient; i++) {
      out.append('I');
      out.append('V');
      num -= 4;
    }

    quotient = num / 1;
    for (int i = 0; i < quotient; i++) {
      out.append('I');
      num -= 1;
    }

    return out.toString();
  }
}

File: 012-IntegerToRoman.js
/**
 * @param {number} num
 * @return {string}
 */
var intToRoman = function(num) {
  var out = "";

  var quotient = Math.floor(num / 1000);
  for (i = 0; i < quotient; i++) {
    out = out.concat("M");
    num -= 1000;
  }

  quotient = Math.floor(num / 900);
  for (i = 0; i < quotient; i++) {
    out = out.concat("CM");
    num -= 900;
  }

  quotient = Math.floor(num / 500);
  for (i = 0; i < quotient; i++) {
    out = out.concat("D");
    num -= 500;
  }

  quotient = Math.floor(num / 400);
  for (i = 0; i < quotient; i++) {
    out = out.concat("CD");
    num -= 400;
  }

  quotient = Math.floor(num / 100);
  for (i = 0; i < quotient; i++) {
    out = out.concat("C");
    num -= 100;
  }

  quotient = Math.floor(num / 90);
  for (i = 0; i < quotient; i++) {
    out = out.concat("XC");
    num -= 90;
  }

  quotient = Math.floor(num / 50);
  for (i = 0; i < quotient; i++) {
    out = out.concat("L");
    num -= 50;
  }

  quotient = Math.floor(num / 40);
  for (i = 0; i < quotient; i++) {
    out = out.concat("XL");
    num -= 40;
  }

  quotient = Math.floor(num / 10);
  for (i = 0; i < quotient; i++) {
    out = out.concat("X");
    num -= 10;
  }

  quotient = Math.floor(num / 9);
  for (i = 0; i < quotient; i++) {
    out = out.concat("IX");
    num -= 9;
  }

  quotient = Math.floor(num / 5);
  for (i = 0; i < quotient; i++) {
    out = out.concat("V");
    num -= 5;
  }

  quotient = Math.floor(num / 4);
  for (i = 0; i < quotient; i++) {
    out = out.concat("IV");
    num -= 4;
  }

  quotient = Math.floor(num / 1);
  for (i = 0; i < quotient; i++) {
    out = out.concat("I");
    num -= 1;
  }

  return out;
};
